chips/lowrisc/src/pinmux.rs

```diff
@@ -0,0 +1,851 @@
+    {
+        MioPadInput::try_from(reg.mio_periph_insel[*self as usize].read(MIO_PERIPH_INSEL::IN_0) as u8)
+    }
+}
+
+/**
+ * Pinmux MIO Output.
+ */
+#[derive(Clone, Copy)]
+#[repr(u8)]
+pub enum MioPadOutput {
+  Ioa0 = 0,
+  Ioa1 = 1,
+  Ioa2 = 2,
+  Ioa3 = 3,
+  Ioa4 = 4,
+  Ioa5 = 5,
+  Ioa6 = 6,
+  Ioa7 = 7,
+  Ioa8 = 8,
+  Iob0 = 9,
+  Iob1 = 10,
+  Iob2 = 11,
+  Iob3 = 12,
+  Iob4 = 13,
+  Iob5 = 14,
+  Iob6 = 15,
+  Iob7 = 16,
+  Iob8 = 17,
+  Iob9 = 18,
+  Iob10 = 19,
+  Iob11 = 20,
+  Iob12 = 21,
+  Ioc0 = 22,
+  Ioc1 = 23,
+  Ioc2 = 24,
+  Ioc3 = 25,
+  Ioc4 = 26,
+  Ioc5 = 27,
+  Ioc6 = 28,
+  Ioc7 = 29,
+  Ioc8 = 30,
+  Ioc9 = 31,
+  Ioc10 = 32,
+  Ioc11 = 33,
+  Ioc12 = 34,
+  Ior0 = 35,
+  Ior1 = 36,
+  Ior2 = 37,
+  Ior3 = 38,
+  Ior4 = 39,
+  Ior5 = 40,
+  Ior6 = 41,
+  Ior7 = 42,
+  //Ior8
+  //Ior9
+  Ior10 = 43,
+  Ior11 = 44,
+  Ior12 = 45,
+  Ior13 = 46,
+}
+
+#[derive(Clone, Copy)]
+#[repr(u8)]
+pub enum MioPeripheralOutput {
+  ConstantZero = 0,
+  ConstantOne = 1,
+  ConstantHighZ = 2,
+  GpioGpio0 = 3,
+  GpioGpio1 = 4,
+  GpioGpio2 = 5,
+  GpioGpio3 = 6,
+  GpioGpio4 = 7,
+  GpioGpio5 = 8,
+  GpioGpio6 = 9,
+  GpioGpio7 = 10,
+  GpioGpio8 = 11,
+  GpioGpio9 = 12,
+  GpioGpio10 = 13,
+  GpioGpio11 = 14,
+  GpioGpio12 = 15,
+  GpioGpio13 = 16,
+  GpioGpio14 = 17,
+  GpioGpio15 = 18,
+  GpioGpio16 = 19,
+  GpioGpio17 = 20,
+  GpioGpio18 = 21,
+  GpioGpio19 = 22,
+  GpioGpio20 = 23,
+  GpioGpio21 = 24,
+  GpioGpio22 = 25,
+  GpioGpio23 = 26,
+  GpioGpio24 = 27,
+  GpioGpio25 = 28,
+  GpioGpio26 = 29,
+  GpioGpio27 = 30,
+  GpioGpio28 = 31,
+  GpioGpio29 = 32,
+  GpioGpio30 = 33,
+  GpioGpio31 = 34,
+  I2c0Sda = 35,
+  I2c0Scl = 36,
+  I2c1Sda = 37,
+  I2c1Scl = 38,
+  I2c2Sda = 39,
+  I2c2Scl = 40,
+  SpiHost1Sd0 = 41,
+  SpiHost1Sd1 = 42,
+  SpiHost1Sd2 = 43,
+  SpiHost1Sd3 = 44,
+  Uart0Tx = 45,
+  Uart1Tx = 46,
+  Uart2Tx = 47,
+  Uart3Tx = 48,
+  PattgenPda0Tx = 49,
+  PattgenPcl0Tx = 50,
+  PattgenPda1Tx = 51,
+  PattgenPcl1Tx = 52,
+  SpiHost1Sck = 53,
+  SpiHost1Csb = 54,
+  FlashCtrlTdo = 55,
+  SensorCtrlAstDebugOut0 = 56,
+  SensorCtrlAstDebugOut1 = 57,
+  SensorCtrlAstDebugOut2 = 58,
+  SensorCtrlAstDebugOut3 = 59,
+  SensorCtrlAstDebugOut4 = 60,
+  SensorCtrlAstDebugOut5 = 61,
+  SensorCtrlAstDebugOut6 = 62,
+  SensorCtrlAstDebugOut7 = 63,
+  SensorCtrlAstDebugOut8 = 64,
+  PwmAonPwm0 = 65,
+  PwmAonPwm1 = 66,
+  PwmAonPwm2 = 67,
+  PwmAonPwm3 = 68,
+  PwmAonPwm4 = 69,
+  PwmAonPwm5 = 70,
+  OtpCtrlTest0 = 71,
+  SysrstCtrlAonBatDisable = 72,
+  SysrstCtrlAonKey0Out = 73,
+  SysrstCtrlAonKey1Out = 74,
+  SysrstCtrlAonKey2Out = 75,
+  SysrstCtrlAonPwrbOut = 76,
+  SysrstCtrlAonZ3Wakeup = 77,
+}
+
+impl TryFrom<u8> for MioPeripheralOutput
+{
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(MioPeripheralOutput::ConstantZero),
+            1 => Ok(MioPeripheralOutput::ConstantOne),
```

Comment:
@cfrantz  please check this enum, maybe you know the simplest way that is 'sound' to convert this.